### PR TITLE
DashR Ideogram

### PR DESCRIPTION
**Pull request by HammadTheOne**
Issue for app: #[issue number here]

# App pull request
- [ ] This is a new app
- [x] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL (new version): https://dash-playground.plotly.host/dashr-ideogram
- Current gallery app URL: https://dash-bio.plotly.host/dash-ideogram/
- Python app repository link: https://github.com/plotly/dash-bio/blob/master/tests/dashbio_demos/app_ideogram.py

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [x] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [ ] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
- [ ] If the app is a reimplementation of a Python gallery app for the
  DashR gallery, the app in this PR mimics, as closely as possible,
  the style and functionality of the existing app.

## The pre-review review

I have addressed all of the following questions:

- [x] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [x] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [x] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

File: apps/dashr-ideogram/assets/general-app-page.css [hunk 1]
```diff
@@ -0,0 +1,257 @@
+@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400');
```
Comment (rpkyle) [inline apps/dashr-ideogram/assets/general-app-page.css]: Could maybe move this to `assets`?

File: apps/dashr-ideogram/app.R [hunk 2]
```diff
@@ -0,0 +1,1162 @@
...
+  
+  
+  listofchromosomes = (sprintf("%d",seq(1:22)))
+  
+  listofchromosomes = c(listofchromosomes, "X")
+  
+  listofchromosomes = c(listofchromosomes, "Y")
+  
+  
+  listOfoptions <- list(
```
Comment (rpkyle) [inline apps/dashr-ideogram/app.R]: I suggest replacing `listOfoptions` with `listOfOptions`, constructed using instead

```
chromosomes <- c(as.character(seq(1:22)), "X", "Y")

listOfOptions <- lapply(chromosomes, function(x) {
  list(label = x, value = x)
})
```

🤓

File: apps/dashr-ideogram/app.R [hunk 3]
```diff
@@ -0,0 +1,1162 @@
...
+  
+      
+    ),
+    
+    'homology' = list(
+      htmlDiv(className = 'app-controls-name', children = 'Chromosomes:' ),
+      dccDropdown(
+        className = 'ideogram-dropdown',
+        id = 'chr-select-1',
+        options = list(
```
Comment (rpkyle) [inline apps/dashr-ideogram/app.R]: Same suggestion as above using the `lapply` syntax 👨‍🎨

File: apps/dashr-ideogram/app.R [hunk 4]
```diff
@@ -0,0 +1,1162 @@
...
+          list('label' = '21', 'value' = '21'),
+          list('label' = '22', 'value' = '22')
+        ) , value = '1'
+      ),
+      
+      dccDropdown(
+        className = 'ideogram-dropdown',
+        id = 'chr-select-2',
+        value = '2',
+        options = list(
```
Comment (rpkyle) [inline apps/dashr-ideogram/app.R]: Same suggestion as above using the `lapply` syntax 👨‍🎨

File: apps/dashr-ideogram/app.R [hunk 5]
```diff
@@ -0,0 +1,1162 @@
...
+    ),
+    
+    'brush' = list(
+      htmlDiv(className = 'app-controls-block', children = list(
+        htmlDiv(className = 'app-controls-name', id = 'brush-control-name', children = 'Chromosome:'),
+        dccDropdown(
+          className = 'ideogram-dropdown',
+          id = 'chr-brush',
+          options = list(
+            list('label' = 'X', 'value' = 'X'),
```
Comment (rpkyle) [inline apps/dashr-ideogram/app.R]: use `lapply` 👨‍🎨

File: apps/dashr-ideogram/app.R [hunk 6]
```diff
@@ -0,0 +1,1162 @@
...
+  
+  
+  
+  # Select which chromosomes to show for the customizable feature.
+  
+   app$callback(
+      output = list(id = 'ideo-custom', property = 'chromosomes'),
+      params = list(input(id = 'displayed-chromosomes', property = 'value')),
+  
+      update_ideogram <- function(value) {
```
Comment (rpkyle) [inline apps/dashr-ideogram/app.R]: Interesting. Why an identity function here? Is this just to keep the same structure as in the other callbacks?

File: apps/dashr-ideogram/app.R [hunk 7]
```diff
@@ -0,0 +1,1162 @@
...
+        annotations_path = NULL
+        annotations_assembly = NULL
+        annotation_tracks = NULL
+        annotation_height = height_input
+        
+        
+        
+        if (annotation_select == 'overlay-1') {
+          annotations_layout = 'overlay'
+          annotations_path = 'https://eweitz.github.io/ideogram/data/annotations/10_virtual_cnvs.json'
```
Comment (rpkyle) [inline apps/dashr-ideogram/app.R]: Could this 🚚 to `assets` folder?

File: apps/dashr-ideogram/app.R [hunk 8]
```diff
@@ -0,0 +1,1162 @@
...
+        }
+        
+        else if (annotation_select == 'histogram') {
+          annotations_path = 'https://eweitz.github.io/ideogram/data/annotations/SRR562646.json'
```
Comment (rpkyle) [inline apps/dashr-ideogram/app.R]: Same, could this 🚚 to `assets` folder?

File: apps/dashr-ideogram/app.R [hunk 9]
```diff
@@ -0,0 +1,1162 @@
...
+          annotations_assembly = 'GRCh37'
+        }  
+        
+        else if (annotations_select == 'overlay-2') {
+          annotations_path = 'https://eweitz.github.io/ideogram/data/annotations/1000_virtual_snvs.json'
```
Comment (rpkyle) [inline apps/dashr-ideogram/app.R]: 🚚 to `assets`? Repeat question for `annotations*` below 🙂

File: apps/dashr-ideogram/assets/general-app-page.css (hunk 1 above)
Comment (HammadTheOne) [reply apps/dashr-ideogram/assets/general-app-page.css]: Done

File: apps/dashr-ideogram/app.R (hunk 2 above)
Comment (HammadTheOne) [reply apps/dashr-ideogram/app.R]: Replaced.

File: apps/dashr-ideogram/app.R (hunk 6 above)
Comment (HammadTheOne) [reply apps/dashr-ideogram/app.R]: Pretty much, just so it's a bit easier for someone who might be new to callbacks.

**Pull request merged**